experimental/dds/tree2/src/shared-tree-core/branch.ts

```diff
@@ -118,6 +118,7 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> exten
 > {
 	public readonly editor: TEditor;
 	private readonly transactions = new TransactionStack<TChange>();
+	private readonly forksInTransactions: Set<SharedTreeBranch<TEditor, TChange>>[] = [];
```

Comment:
I was wondering about that. I vaguely remember hearing that we might get rid of the existing `TransactionStack`, in which case adding stuff to it that we will want to keep seems like extra pain for no reason. @noencke - you probably have more context on that question so whatever you say goes.

Reply:
Yea, I'm not totally sure, but since we already have a stack right now it seems confusing to make a second stack. I think if we keep it consolidated it will be clearer what to do later if we do decide to try a simplification.

We technically don't need a stack, either. We could say that for _inner_ transactions (i.e. a transaction that starts within another transaction), the disposal of any branches created happens after the _outer_ transaction closes, not after the _inner_ transaction closes. That would be simpler to implement and just as correct, however, maybe it's a little less "precise" in terms of the behavior that a user experiences. It's probably best to have the behavior be consistent for all levels of nesting lest the user get used to not worrying about disposal on an inner transaction and then someday be surprised when the same approach doesn't work for an outer transaction. So, I say keep the stack for this feature even though it's a little annoying to keep track of.